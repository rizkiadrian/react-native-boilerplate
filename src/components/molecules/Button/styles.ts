import {StyleSheet} from 'react-native';
import stylesVar from 'src/global-styles/stylesVar';

export default StyleSheet.create({
  root: {
    width: 155,
    height: 25,
    borderRadius: 16,
  },
  touchable: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontFamily: stylesVar.fontFamily(600),
    color: stylesVar.colorVariable.white,
    fontSize: 16,
    lineHeight: 24,
  },
});
