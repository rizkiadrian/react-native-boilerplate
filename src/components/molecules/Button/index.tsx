import React from 'react';
import {
  GestureResponderEvent,
  StyleProp,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import stylesVar from 'src/global-styles/stylesVar';
import styles from 'src/components/molecules/Button/styles';
import Text from 'src/components/atoms/Text';

interface IButton {
  label?: string;
  onPress?: ((event: GestureResponderEvent) => void) | undefined;
  rootStyle?: StyleProp<ViewStyle> | undefined;
}

function Button(props: IButton): JSX.Element {
  return (
    <LinearGradient
      colors={stylesVar.gradient.blue.colors}
      useAngle
      angle={stylesVar.gradient.blue.degree}
      angleCenter={{x: 0.5, y: 0.5}}
      style={[styles.root, props.rootStyle]}>
      <TouchableOpacity style={styles.touchable} onPress={props.onPress}>
        <Text style={styles.text}>{props.label}</Text>
      </TouchableOpacity>
    </LinearGradient>
  );
}

export default Button;
