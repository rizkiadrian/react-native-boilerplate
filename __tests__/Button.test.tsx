/**
 * Mock React Native Linear Gradient due to compiling error on jest
 */
jest.mock('react-native-linear-gradient', () => 'LinearGradient');

import React from 'react';
import Button from 'src/components/molecules/Button';
import renderer from 'react-test-renderer';

it(`src/components/molecules/Button Component render correctly`, () => {
  const tree = renderer.create(<Button label="dream" />);
  expect(tree).toMatchSnapshot();
});
