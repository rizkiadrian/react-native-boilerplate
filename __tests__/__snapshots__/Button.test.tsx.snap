// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/molecules/Button Component render correctly 1`] = `
<LinearGradient
  angle={271}
  angleCenter={
    {
      "x": 0.5,
      "y": 0.5,
    }
  }
  colors={
    [
      "#176FF2",
      "#196EEE",
    ]
  }
  style={
    [
      {
        "borderRadius": 16,
        "height": 25,
        "width": 155,
      },
      undefined,
    ]
  }
  useAngle={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#3A544F",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 14,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
    >
      dream
    </Text>
  </View>
</LinearGradient>
`;
